### ISSUE TITLE
Exception causing Python HTTP server to crash

### ISSUE THREAD

**Issue by redt3am** [EmpireProject/Empire#446]
## Empire Version
Latest master branch

## OS Information (Linux flavor, Python version)
Ubuntu 16.04, python 2.7.12

## Expected behavior and description of the error, including any actions taken immediately prior to the error. The more detail the better.
I get this error after removing some active agents without killing. Perhaps it is from removing an agent without killing it, causing it to continue to callback with invalid parameters?

I should also note that Empire tends to stop working when having many clients for some reason. The error does not occur, but the python web server will stop working silently while the interface shows no warning, only nothing will be able to connect. 

## Screenshot of error, embedded text output, or Pastebin link to the error
````
----------------------------------------
Exception happened during processing of request from ('xx.xx.xx.xx', xxxxx)
Traceback (most recent call last):
  File "/usr/lib/python2.7/SocketServer.py", line 290, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/usr/lib/python2.7/SocketServer.py", line 318, in process_request
    self.finish_request(request, client_address)
  File "/usr/lib/python2.7/SocketServer.py", line 331, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib/python2.7/SocketServer.py", line 652, in __init__
    self.handle()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 340, in handle
    self.handle_one_request()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 310, in handle_one_request
    self.raw_requestline = self.rfile.readline(65537)
  File "/usr/lib/python2.7/socket.py", line 480, in readline
    data = self._sock.recv(self._rbufsize)
error: [Errno 104] Connection reset by peer
----------------------------------------
````

## Any additional information
This causes the server to crash, which causes it to stop communicating with agents.

**Comment by killswitch-GUI** [comment]
this traceback shows a issue in the Python core, this may be a bug with Pythons SocketServer. Let me see what we can do to prevent an entire crash.

**Comment by redt3am** [comment]
Thank you! What about the server crashing during heavier loads and over time? Sometimes it happens with ~10 agents. It will work for a few hours, then it suddenly become unresponsive. The UI is fine, but the python server stops working, and I don't think it ever become responsive again. When loading the url, there is a timeout. Could it be because of thread limits? I will look into it because it is very annoying to have to restart the Empire every time this happens, and it can cause you to miss agents. With fewer agents it lasts much longer sometimes a day, but eventually it could stop working.

**Comment by killswitch-GUI** [comment]
@redt3am we really don't know right now. *not a good answer

**Comment by xorrior** [comment]
I believe this issue should be fixed once matter is merged with 2.0 beta since we now use flask
